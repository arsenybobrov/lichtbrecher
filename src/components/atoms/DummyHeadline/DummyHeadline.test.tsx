import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import DummyHeadline, { HeadlineType } from './DummyHeadline';

// eslint-disable-next-line import/prefer-default-export
export const headingMock = (level: number) => ({
  type: `heading${level}` as HeadlineType,
  text: `this is a h${level} headline`,
});

describe('nextLink snapshot', () => {
  it.each`
    name             | mock
    ${'h1 headline'} | ${headingMock(1)}
    ${'h2 headline'} | ${headingMock(2)}
    ${'h3 headline'} | ${headingMock(3)}
    ${'h4 headline'} | ${headingMock(4)}
    ${'h5 headline'} | ${headingMock(5)}
    ${'h6 headline'} | ${headingMock(6)}
  `('$name', ({ mock }) => {
  const tree = renderer.create(<DummyHeadline {...mock} />).toJSON();
  expect(tree).toMatchSnapshot();
});
});
