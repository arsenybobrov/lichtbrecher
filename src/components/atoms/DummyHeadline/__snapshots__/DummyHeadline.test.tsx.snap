// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`nextLink snapshot h1 headline 1`] = `
<h1>
  this is a h1 headline
</h1>
`;

exports[`nextLink snapshot h2 headline 1`] = `
<h2>
  this is a h2 headline
</h2>
`;

exports[`nextLink snapshot h3 headline 1`] = `
<h3>
  this is a h3 headline
</h3>
`;

exports[`nextLink snapshot h4 headline 1`] = `
<h4>
  this is a h4 headline
</h4>
`;

exports[`nextLink snapshot h5 headline 1`] = `
<h5>
  this is a h5 headline
</h5>
`;

exports[`nextLink snapshot h6 headline 1`] = `
<h6>
  this is a h6 headline
</h6>
`;
